ipc/loader.py

```diff
@@ -0,0 +1,37 @@
+#
+#  Copyright (C) 2017, Jaguar Land Rover
+#
+#  This program is licensed under the terms and conditions of the
+#  Mozilla Public License, version 2.0.  The full text of the 
+#  Mozilla Public License is at https://www.mozilla.org/MPL/2.0/
+#
+
+import inspect
+import importlib
+
+def _load_module(modulename):
+    try:
+        return importlib.import_module(modulename)
+    except ImportError:
+        print("Error loading module: {}".format(modulename))
+        raise
+
+def _exist_method(module, method):
+    return hasattr(module, method) and inspect.isroutine(getattr(module, method))
+
+def load_plugin(modulename):
+    # Load plugin module.
+    module = _load_module(modulename)
```

Comment:
I think you should catch the exception, report the error and exit the program cleanly instead of letting the exception stop it.